libcamera_app.cpp

```diff
@@ -40,27 +43,27 @@ void LibcameraApp::OpenCamera()
 {
 	// Make a preview window.
 	if (options_->nopreview)
-		preview_ = std::make_unique<NullPreview>(options_.get());
+		preview_ = std::unique_ptr<Preview>(make_null_preview(options_.get()));
 	else
 	{
 		try
 		{
-			preview_ = std::make_unique<EglPreview>(options_.get());
+			preview_ = std::unique_ptr<Preview>(make_egl_preview(options_.get()));
 			if (options_->verbose)
 				std::cout << "Made X/EGL preview window" << std::endl;
 		}
 		catch (std::exception const &e)
 		{
 			try
 			{
-				preview_ = std::make_unique<DrmPreview>(options_.get());
+				preview_ = std::unique_ptr<Preview>(make_drm_preview(options_.get()));
 				if (options_->verbose)
 					std::cout << "Made DRM preview window" << std::endl;
 			}
 			catch (std::exception const &e)
 			{
 				std::cout << "Preview window unavailable" << std::endl;
-				preview_ = std::make_unique<NullPreview>(options_.get());
+				preview_ = std::unique_ptr<Preview>(make_null_preview(options_.get()));
 			}
 		}
 	}
```

Comment:
I'm curious why we added make_xxx_preview() functions over the existing std::make_unique mechanism.  Is it that we have no reference to xxxPrieview class definitions now, or am I missing something else?

Reply:
Yes, it's so that libcamera_app.cpp doesn't have to drag in constructors for NullPreview, EglPreview etc.

Reply:
Got it!